CMakeLists.txt

```diff
@@ -822,20 +822,20 @@ set(ZLIB_PUBLIC_HDRS
 set(ZLIB_PRIVATE_HDRS
     adler32_p.h
     chunkset_tpl.h
-    crc32.h
     crc32_p.h
+    crc32_tbl.h
     deflate.h
     deflate_p.h
     functable.h
     gzguts.h
     inffast.h
-    inffixed.h
+    inffixed_tbl.h
     inflate.h
     inflate_p.h
     inftrees.h
     insert_string_tpl.h
     match_tpl.h
-    trees.h
+    trees_tbl.h
     trees_emit.h
     trees_p.h
```

Comment:
Maybe rename `trees_p.h` to `trees.h` too? It was named that because the `trees.h` name was already taken.

Reply:
Point of having _p suffix in header basename is that it discourages external code to use it when including zlib-ng as subproject.

Reply:
FYI These filenames are not in alphabetical order anymore...

Reply:
trees_p.h contains the bits that are not private to trees.c, and it needs to be included from the tool maketrees.c currently.
So the name is in my opinion wrong and breaks with how we treat the _p file naming scheme in the rest of the codebase.

Also if we wanted to discourage external users, then almost all our header files would need to have the _p suffix.

Reply:
If trees_p.h contains non-private bits, they should be moved to trees.h instead...

The point of having _p suffix only applies when there is two files, one without the suffix, and one with the suffix.

Reply:
That is the point I am making. Now that there is no trees.h, trees_p.h can finally be renamed to trees.h like it should have been. The only thing stopping that was that trees.h was an auto-generated file for the tables.

Reply:
@Dead2 I'm not even trying to understand all the complex details... As long as we agree there is still something to fix, then everything is good...

Reply:
I have renamed trees_p.h to trees.h and also fixed the alphabetical ordering. 
@Dead2 should crc32_p.h be renamed crc32.h and adler32_p.h be renamed adler32.h?

Reply:
Well i have done it, I can remove the commits if necessary.

Reply:
Well, how we used it in the past has been to have `_p.h` contain C functions that needs to live in a header file for whatever reason. And .h for defines, function declarations and tables.
`crc32_p.h` follows that way of doing it, but `adler32_p.h` has both functions and defines.

So should `adler32_p.h` be split or should we change how we do this? IDK, I am open for suggestions.
`_tpl.h` and `_tbl.h` are both great.
`_p.h` is maybe not, should we put "included functions" into a file with a a new and better suffix perhaps? `_func.h` comes to mind, would that be a good place to put them?

Another alternative is using `_p.h` for all non-externally visible header files, but would require renaming almost all header files, and I honestly don't think that solves anything for us, and I think most programmers are able to figure out they need to include zlib-ng.h.

Open to suggestions, but my vote is currently on `_func.h` (or a better name if someone comes up with one).

Reply:
I see it as *_p.h having everything that is not/should not be externally visible but shared with multiple files in our source tree, including defines and static inline functions. This keeps us safe when external project has some "global" define that might conflict with what we use, as it will trigger an error/warning during subproject build.

Like I said already earlier, we should only use _p suffix when there is already header file with same basename but without the suffix... Having it on all non-public header files serve no purpose.